app/main.py

```diff
@@ -1,13 +1,23 @@
 def format_linter_error(error: dict) -> dict:
-    # write your code here
-    pass
+    return {
+        "column": error["column_number"],
+        "line": error["line_number"],
+        "message": error["text"],
+        "name": error["code"],
+        "source": "flake8"
+    }
 
 
 def format_single_linter_file(file_path: str, errors: list) -> dict:
-    # write your code here
-    pass
+    return {
+        "errors": [format_linter_error(error) for error in errors],
+        "path": file_path,
+        "status": "failed" if any(errors) else "passed"
+    }
 
 
 def format_linter_report(linter_report: dict) -> list:
-    # write your code here
-    pass
+    return [
+        format_single_linter_file(path_name, list_of_errors)
+        for path_name, list_of_errors in linter_report.items()
+    ]
```

Comment:
It is better to break the line with generators in such way:
[
some variable
for some variable
in some iterable
]

Reply:
@YuriiKindrat     return [
        format_single_linter_file(path_name, list_of_errors)
        for path_name, list_of_errors in linter_report.items()
    ]   in this variant there are no mistakes  in flake8 app, if i use your variant , i  have 2 mistakes, so i did another way. Watch please.